inc/css/class-css-handler.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+namespace ThemeIsle\BlockCSS;
+
+use ThemeIsle\BlockCSS;
+
+use tubalmartin\CssMin\Minifier as CSSmin;
+
+/**
+ * Class CSS_Handler
+ */
+class CSS_Handler extends BlockCSS {
+
+	/**
+	 * The main instance var.
+	 *
+	 * @var CSS_Handler
+	 */
+	public static $instance = null;
+
+	/**
+	 * Rest route namespace.
+	 *
+	 * @var CSS_Handler
+	 */
+	public $namespace = 'themeisle-gutenberg-blocks/';
+
+	/**
+	 * Rest route version.
+	 *
+	 * @var CSS_Handler
+	 */
+	public $version = 'v1';
+
+	/**
+	 * Constructor function for the module.
+	 *
+	 * @method __construct
+	 */
+	public function __construct() {
+		parent::__construct();
+	}
+
+	/**
+	 * Initialize the class
+	 */
+	public function init() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+		add_action( 'rest_api_init', array( $this, 'autoload_block_classes' ) );
+		add_action( 'before_delete_post', array( $this, 'delete_css_file' ) );
+	}
+
+	/**
+	 * Register REST API route
+	 */
+	public function register_routes() {
+		$namespace = $this->namespace . $this->version;
+
+		register_rest_route(
+			$namespace,
+			'/save_post_meta/(?P<id>\d+)',
+			array(
+				array(
+					'methods'	=> \WP_REST_Server::EDITABLE,
+					'callback'	=> array( $this, 'save_post_meta' ),
+					'args'		=> array(
+						'id'	=> array(
+							'type'        => 'intval',
+							'required'    => true,
+							'description' => __( 'ID of the Post.', 'textdomain' ),
+							'validate_callback' => function( $param, $request, $key ) {
+								return is_numeric( $param );
+							}
+						),
+					),
+				),
+			)
+		);
+
+		register_rest_route(
+			$namespace,
+			'/save_block_meta/(?P<id>\d+)',
+			array(
+				array(
+					'methods'	=> \WP_REST_Server::EDITABLE,
+					'callback'	=> array( $this, 'save_block_meta' ),
+					'args'		=> array(
+						'id'	=> array(
+							'type'        => 'intval',
+							'required'    => true,
+							'description' => __( 'ID of the Reusable Block.', 'textdomain' ),
+							'validate_callback' => function( $param, $request, $key ) {
+								return is_numeric( $param );
+							}
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Function to save post CSS.
+	 *
+	 * @return array|bool|\WP_Error
+	 */
+	public function save_post_meta( \WP_REST_Request $request ) {
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return false;
+		}
+
+		$post_id = $request->get_param( 'id' );
+		$css = $this->get_blocks_css( $post_id );
+		$css = wp_filter_nohtml_kses( $css );
+
+		if ( ! empty( $css ) ) {
+			$compressor = new CSSmin;
+
+			// Override any PHP configuration options before calling run()
```

Comment:
this part seems the same with this one https://github.com/Codeinwp/gutenberg-blocks/pull/119/files#diff-a7584dc33a3555e137d425208e6b1a67R162-R170 . We can DRY the code the create a single method, routine for compression.

Reply:
Fixed https://github.com/Codeinwp/gutenberg-blocks/pull/119/commits/0488851e4bd051c0f90104b990ef0f226e13f4a5